curves/bn254/utils.go

```diff
@@ -26,19 +27,36 @@ func CopyPointsToDevice(points []bn254.G1Affine, pointsBytes int, copyDone chan
 		devicePtr, _ := goicicle.CudaMalloc(pointsBytes)
 		iciclePoints := BatchConvertFromG1Affine(points)
 		goicicle.CudaMemCpyHtoD[icicle.G1PointAffine](devicePtr, iciclePoints, pointsBytes)
-		
+
 		copyDone <- devicePtr
 	}
 }
 
+// TODO find better way to handle
+func CopyScalarsToHost(a_device unsafe.Pointer, n int, sizeBytes int) []fr.Element {
+	outHost := make([]fr.Element, n)
+
+	G := new(errgroup.Group)
+	G.Go(func() (err error) {
+		ret := goicicle.CudaMemCpyDtoH[fr.Element](outHost, a_device, sizeBytes)
+		if ret != 0 {
+			err = fmt.Errorf("Memory copy error")
+		}
+		return
+	})
+	G.Wait()
```

Comment:
Whats the reason for using goroutine and Wait for a single operation with no other parallel operations?

Reply:
Updated to remove.